src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -942,25 +944,231 @@ private void DecodeModuleInitializerAttribute(DecodeWellKnownAttributeArguments<
+        {
+            var diagnostics = (BindingDiagnosticBag)arguments.Diagnostics;
+            if (version != 1)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationUnsupportedVersion, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 0));
+                return;
+            }
+
+            if (data is null)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 1));
+                return;
+            }
+
+            byte[] bytes;
+            try
+            {
+                bytes = Convert.FromBase64String(data);
+            }
+            catch (FormatException)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 1));
+                return;
+            }
+
+            // format:
+            // - 16 bytes of target file content hash (xxHash128)
+            // - int32 position (little endian)
+            // - utf-8 display filename
+            const int hashIndex = 0;
+            const int hashSize = 16;
+            const int positionIndex = hashIndex + hashSize;
+            const int positionSize = sizeof(int);
+            const int displayNameIndex = positionIndex + positionSize;
+            const int minLength = displayNameIndex;
+
+            if (bytes.Length < minLength)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 1));
+                return;
+            }
+
+            var hash = bytes.AsMemory(start: hashIndex, length: hashSize);
+            var position = BinaryPrimitives.ReadInt32LittleEndian(bytes.AsSpan(start: positionIndex));
+
+            string displayFileName;
+            try
+            {
+                displayFileName = Encoding.UTF8.GetString(bytes, index: displayNameIndex, count: bytes.Length - displayNameIndex);
+            }
+            catch (ArgumentException)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 1));
+                return;
+            }
+
+            var attributeSyntax = arguments.AttributeSyntaxOpt;
+            Debug.Assert(attributeSyntax is object);
+            var attributeLocation = attributeSyntax.Location;
+
+            var interceptorsNamespaces = ((CSharpParseOptions)attributeSyntax.SyntaxTree.Options).InterceptorsPreviewNamespaces;
+            var thisNamespaceNames = getNamespaceNames();
+            var foundAnyMatch = interceptorsNamespaces.Any(ns => isDeclaredInNamespace(thisNamespaceNames, ns));
+            if (!foundAnyMatch)
+            {
+                reportFeatureNotEnabled(diagnostics, attributeSyntax, thisNamespaceNames);
```

Comment:
To reinforce the point about test coverage, this code doesn't seem covered.

Reply:
Have ported/added tests for all the newly added code paths.